exchanges/gateio/gateio_test.go

```diff
@@ -3580,3 +3633,43 @@ func TestGetIntervalString(t *testing.T) {
 	_, err = getIntervalString(kline.FiveDay)
 	assert.ErrorIs(t, err, kline.ErrUnsupportedInterval, "Any other random interval should also be invalid")
 }
+
+func TestWebsocketSubmitOrders(t *testing.T) {
+	t.Parallel()
+
+	_, err := e.WebsocketSubmitOrders(t.Context(), nil)
+	require.ErrorIs(t, err, asset.ErrNotSupported)
+
+	sub := &order.Submit{
+		Exchange:    e.Name,
+		AssetType:   asset.Spot,
+		Side:        order.Buy,
+		Type:        order.Market,
+		QuoteAmount: 10,
+	}
+	_, err = e.WebsocketSubmitOrders(t.Context(), []*order.Submit{sub})
+	require.ErrorIs(t, err, order.ErrPairIsEmpty)
+
+	sub.Pair = currency.NewBTCUSD()
+	cpy := *sub
+	cpy.AssetType = asset.Futures
+	_, err = e.WebsocketSubmitOrders(t.Context(), []*order.Submit{sub, &cpy})
+	require.ErrorIs(t, err, errSingleAssetRequired)
+
+	cpy.AssetType = asset.Spread
+	sub.AssetType = asset.Spread
+	_, err = e.WebsocketSubmitOrders(t.Context(), []*order.Submit{sub, &cpy})
+	require.ErrorIs(t, err, asset.ErrNotSupported)
+
+	sub.AssetType = asset.USDTMarginedFutures
+	cpy.AssetType = asset.USDTMarginedFutures
+	_, err = e.WebsocketSubmitOrders(t.Context(), []*order.Submit{sub, &cpy})
+	require.ErrorIs(t, err, common.ErrNotYetImplemented)
+
+	sharedtestvalues.SkipTestIfCredentialsUnset(t, e, canManipulateRealOrders)
+
+	sub.AssetType = asset.Spot
+	cpy.AssetType = asset.Spot
+	_, err = e.WebsocketSubmitOrders(t.Context(), []*order.Submit{sub, &cpy})
+	require.NoError(t, err)
```

Comment:
Only thing I encountered here is getting a "websocket is not connected" error. Is there any additional steps required that you need to add to this part of the test to have it run properly?